app/volume.go

```diff
@@ -274,12 +346,35 @@ func (job *Job) backupAndCleanup() (err error) {
 	backupAPI := job.api.BackupVolume
 	volumeAPI := job.api.Volume
 	snapshot := job.snapshotName
-	volume := job.volume
-	volumeName := volume.Name
+	volumeName := job.volumeName
+
 	defer func() {
 		err = errors.Wrapf(err, "failed to complete backupAndCleanup for %v", volumeName)
 	}()
 
+	volume, err := volumeAPI.ById(volumeName)
+	if err != nil {
+		return errors.Wrapf(err, "could not get volume %v", volumeName)
+	}
+
+	// Check the size of VolumeHead, if it is empty, skip the recurring backup.
+	// We don't want to overwrite the old backups with new identical backups.
+	// This happens when the volume is detached for a long time and has no new data.
+	volumeHead, err := volumeAPI.ActionSnapshotGet(volume, &longhornclient.SnapshotInput{
```

Comment:
Separate the volume head size check into another commit. It will affect all the recurring jobs, not only detached one.